### PR TITLE
Add tensorboard and check in vendor for the code search example.

### PR DESCRIPTION
**Pull request by jlewi**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/255)

### CODE REVIEW

**Comment by activatedgeek** [comment]
/lgtm

It appears that most changes to the Ksonnet files are just reformatting

File: code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet [hunk 1]
```diff
@@ -0,0 +1,143 @@
+// @apiVersion 0.1
```
Comment (inc0) [inline code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet]: should this be in different PR?

File: code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-job-simple-v1alpha1.jsonnet [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// @apiVersion 0.1
```
Comment (inc0) [inline code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-job-simple-v1alpha1.jsonnet]: ditto, seems like component from different problem

File: code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-serving-with-istio.jsonnet [hunk 3]
```diff
@@ -0,0 +1,179 @@
+// @apiVersion 0.1
```
Comment (inc0) [inline code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-serving-with-istio.jsonnet]: If you want to add tf-serving + istion component, let's make dedicated PR

File: code_search/kubeflow/components/params.libsonnet [hunk 4]
```diff
@@ -79,5 +79,10 @@
       indexFile: $.components['t2t-code-search'].workingDir + '/code_search_index.nmslib',
       servingUrl: 'http://t2t-code-search.kubeflow:9001/v1/models/t2t-code-search:predict',
     },
+    tensorboard: {
+      image: "tensorflow/tensorflow:1.8.0",
+      logDir: "gs://cloud-ml-dev_jlewi_cs/mode/2018_0825/",
```
Comment (inc0) [inline code_search/kubeflow/components/params.libsonnet]: do you want to keep your personal path here?

File: code_search/kubeflow/components/t2t-job.libsonnet [hunk 5]
```diff
@@ -84,7 +84,7 @@ local baseParams = std.extVar("__ksonnet/params").components["t2t-job"];
     local workerEnv = [
       {
         name: "GOOGLE_APPLICATION_CREDENTIALS",
-        value: "/secret/gcp-credentials/user-gcp-sa.json"
+        value: "/secret/gcp-credentials/user-gcp-sa.json",
```
Comment (inc0) [inline code_search/kubeflow/components/t2t-job.libsonnet]: can you add another env variable where you point to log dir? Then we could refer to it in code and allow everyone to use tensorboard with this example

File: code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet (hunk 1 above)
Comment (jlewi) [reply code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet]: I'm not following? You want one PR for adding vendor and one for adding TB?

Comment (jlewi) [reply code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet]: The TensorBoard prototoype comes from the examples package. So we need to install the entire examples package in order to get TB which brings in other prototypes as well.

File: code_search/kubeflow/components/t2t-job.libsonnet (hunk 5 above)
Comment (jlewi) [reply code_search/kubeflow/components/t2t-job.libsonnet]: Can you explain what you are asking? Everyone can use tensorboard with this example. They just configure their TB job to point at the same directory that their T2T job writes the model.

**Comment by jlewi** [comment]
Thanks PTAL

File: code_search/kubeflow/components/t2t-job.libsonnet (hunk 5 above)
Comment (inc0) [reply code_search/kubeflow/components/t2t-job.libsonnet]: I was thinking of allowsing something like `tf.summary.FileWriter(os.getenv("KUBEFLOW_LOGDIR"), sess.graph))`

File: code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet (hunk 1 above)
Comment (inc0) [reply code_search/kubeflow/vendor/kubeflow/examples@defc235463799d5600001ee0ed6ef68f7af24a17/prototypes/tf-batch-predict.jsonnet]: ok, makes sense, didn't notice we add whole component

File: code_search/kubeflow/components/t2t-job.libsonnet (hunk 5 above)
Comment (jlewi) [reply code_search/kubeflow/components/t2t-job.libsonnet]: Why do we need that? The trainer code takes command line argument modelDir which controls where the model and event files are written. So if the user wants to change the location of the event files. They can just change that.

**Comment by jlewi** [comment]
@inc0 Ping?

File: code_search/kubeflow/components/t2t-job.libsonnet (hunk 5 above)
Comment (inc0) [reply code_search/kubeflow/components/t2t-job.libsonnet]: ahh that makses sense. I've missed that.

**Comment by jlewi** [comment]
Thanks
/approve
